### PR TITLE
add pip version override for virtualenv in nox

### PR DESCRIPTION
**Pull request by leahecole**
## Description

Needed in #5782 

Note: It's a good idea to open an issue first for discussion.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: noxfile_config.py [hunk 1]
```diff
@@ -32,6 +32,8 @@
     # to use your own Cloud project.
     "gcloud_project_env": "GOOGLE_CLOUD_PROJECT",
     # 'gcloud_project_env': 'BUILD_SPECIFIC_GCLOUD_PROJECT',
+    # If you need to use a specific version of pip, change _pip_version_override to the string representation of the version number, for example, "20.2.4"
```
Comment (dandhlee) [inline noxfile_config.py:35]: Lint _might_ complain saying this is too long :C

**Review by dandhlee — APPROVED** [review]
LGTM! 😄

File: noxfile_config.py (hunk 1 above)
Comment (leahecole) [reply noxfile_config.py:35]: haha I'm fixing it now even though it didn't complain

**Comment by leahecole** [comment]
sry I'm making a git mess w/ noxfile_configs I'll fix it in a sec

File: noxfile_config.py (hunk 1 above)
Comment (dandhlee) [reply noxfile_config.py:35]: Sounds good! XD

**Comment by leahecole** [comment]
> sry I'm making a git mess w/ noxfile_configs I'll fix it in a sec

Fixed

**Review by engelke — APPROVED** [review]

**Pull request merged**